### PR TITLE
Fix echo2 test

### PR DESCRIPTION
**Pull request by dio**
This patch fixes the `echo2` network filter integration test. By actually testing `echo2` instead of `envoy.echo` filter.

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: echo2_config.cc [hunk 1]
```diff
@@ -16,8 +16,20 @@ class Echo2ConfigFactory : public NamedNetworkFilterConfigFactory {
 public:
   // NamedNetworkFilterConfigFactory
   NetworkFilterFactoryCb createFilterFactory(const Json::Object&, FactoryContext&) override {
-    return [](Network::FilterManager& filter_manager)
-        -> void { filter_manager.addReadFilter(Network::ReadFilterSharedPtr{new Filter::Echo2()}); };
+    return [](Network::FilterManager& filter_manager) -> void {
```
Comment (htuch) [inline echo2_config.cc]: Do we still need v1 JSON here? Best to start purging this from the examples.

Comment (dio) [reply echo2_config.cc]: Done at 6368076. To clarify, this incremental update of `echo2` test is actually to support https://github.com/envoyproxy/envoy/pull/2894, i.e. to remove the ci noise regarding this repo.

File: echo2_server.yaml [hunk 2]
```diff
@@ -18,12 +18,12 @@ static_resources:
         address: 127.0.0.1
         port_value: 0
     filter_chains:
-      filters:
-        name: envoy.ratelimit
+    - filters:
+      - name: envoy.ratelimit
```
Comment (mattklein123) [inline echo2_server.yaml]: Can we kill the ratelimit filter also? I think this is also going to cause issues for how the integration/extensions will be setup after all of my changes.

Comment (dio) [reply echo2_server.yaml]: Killed. 152055e

**Pull request merged**